be/src/exec/vectorized/aggregate/agg_hash_map.h

```diff
@@ -60,6 +60,32 @@ using SliceAggTwoLevelHashMap =
         phmap::parallel_flat_hash_map<Slice, AggDataPtr, SliceHashWithSeed<seed>, SliceEqual,
                                       phmap::priv::Allocator<phmap::priv::Pair<const Slice, AggDataPtr>>, PHMAPN>;
 
+static_assert(sizeof(AggDataPtr) == sizeof(size_t));
+#define PRECOMPUTE_HASH_VALUES_NUMBER(column, prefetch_dist)             \
+    size_t const column_size = column->size();                           \
+    size_t* hash_values = reinterpret_cast<size_t*>(agg_states->data()); \
```

Comment:
`agg_states->data()` will be rewritten by `iter->second;` later?

Reply:
Yes, you got it. Here is a small trick, that `agg_states` is both return result, and hash value array. This array has been reused. But reuse has positive effect, that this array is almost in cache everytime.